packages/terra-hookshot/src/Hookshot.jsx

```diff
@@ -0,0 +1,335 @@
+      }
+      this.cachedRects.targetRect = HookshotUtils.getBounds(this.props.targetRef());
+    }
+
+    this.content.rect = this.cachedRects.contentRect;
+    this.target.rect = this.cachedRects.targetRect;
+
+    const result = HookshotUtils.positionStyleFromBounds(
+      this.cachedRects.boundingRect,
+      this.content,
+      this.target,
+      this.props.attachmentMargin,
+      this.props.attachmentBehavior,
+    );
+
+    let styleUpdated = false;
+    if (this.contentNode.style.position !== result.style.position) {
+      this.contentNode.style.position = result.style.position;
+      styleUpdated = true;
+    }
+    const newTransform = `translate3d(${result.style.left}, ${result.style.top}, 0px)`;
+    if (this.contentNode.style.transform !== newTransform) {
+      this.contentNode.style.transform = newTransform;
+      styleUpdated = true;
+    }
+    if (this.contentNode.style.opacity !== '1') {
+      this.contentNode.style.opacity = '1';
+      styleUpdated = true;
+    }
+    if (styleUpdated) {
+      this.cachedRects.contentRect = HookshotUtils.getBounds(this.contentNode);
+      result.positions.content.rect = this.cachedRects.contentRect;
+    }
+
+    if (this.props.onPosition) {
+      this.props.onPosition(
+        event,
+        result.positions,
+      );
+    }
+  }
+
+  destroy() {
+    this.disableListeners();
+    this.contentNode = null;
+  }
+
+  update(event) {
+    if (!this.props.targetRef() || !this.contentNode) {
+      return;
+    }
+    this.updateHookshot(event);
+  }
+
+  updateHookshot(event) {
+    const resetCache = !event || (event.type !== 'scroll' && event.type !== 'touchmove');
+    this.position(event, resetCache);
+  }
+
+  cloneContent(content) {
+    return React.cloneElement(content, { refCallback: this.wrappedRefCallback(content) });
+  }
+
+  wrappedRefCallback(content) {
+    const initialRefCallback = content.props.refCallback;
+
+    return (node) => {
+      this.setContentNode(node);
+
+      if (initialRefCallback) {
+        initialRefCallback(node);
+      }
+    };
+  }
+
+  render() {
+    /* eslint-disable no-unused-vars */
+    const {
+      attachmentBehavior,
+      attachmentMargin,
+      boundingRef,
+      children,
+      contentAttachment,
+      contentOffset,
+      isEnabled,
+      isOpen,
+      targetRef,
+      targetAttachment,
+      targetOffset,
+      onPosition,
+    } = this.props;
+    /* eslint-enable no-unused-vars */
+    if (!isOpen) {
+      return null;
+    }
+
+    this.content = {
+      offset: HookshotUtils.parseOffset(contentOffset),
+      attachment: HookshotUtils.parseAttachment(contentAttachment),
+    };
+    this.target = {
+      offset: HookshotUtils.parseOffset(targetOffset),
+      attachment: targetAttachment ? HookshotUtils.parseAttachment(targetAttachment) : HookshotUtils.mirrorAttachment(this.content.attachment),
+    };
+
+    if (document.getElementsByTagName('html')[0].getAttribute('dir') === 'rtl') {
```

Comment:
Taking the directionality as a prop instead would make this more flexible (unnecessarily so?)